crates/node-core/src/args/network.rs

```diff
@@ -169,17 +169,6 @@ impl NetworkArgs {
                         .build(),
                 )
             })
-            // apply discovery settings
-            .apply(|builder| {
-                let rlpx_socket = (self.addr, self.port).into();
-                let mut builder = self.discovery.apply_to_builder(builder, rlpx_socket);
```

Comment:
not so sure about removing this call because we still need this for discovery settings I believe,

perhaps this explains the failing test

Reply:
failing test is flaky since a while <https://github.com/paradigmxyz/reth/issues/8166>

Reply:
I don't think removing this call outright is correct here, please check

Reply:
I see, maybe keep `.apply` and only remove `if chain_spec.is_optimism() ...` part